docs/src/pages/guide.md

```diff
@@ -0,0 +1,94 @@
+# Getting Started
+
+`Nabla.jl` has two interfaces, both of which we expose to the end user. We first provide a minimal working example with the low-level interface, and subsequently show how the high-level interface can be used to achieve similar results. More involved examples can be found [here](https://github.com/invenia/Nabla.jl/tree/master/examples).
+
+## A Toy Problem
+
+Consider the gradient of a vector-quadratic function. The following code snippet constructs such a function, and inputs `x` and `y`.
+```@example toy
+using Nabla
+
+# Generate some data.
+rng, N = MersenneTwister(123456), 2
+x, y = randn.(rng, [N, N])
+A = randn(rng, N, N)
+
+# Construct a vector-quadratic function in `x` and `y`.
+f(x, y) = y.' * (A * x)
+f(x, y)
+```
+
+Only a small amount of [matrix calculus](https://en.wikipedia.org/wiki/Matrix_calculus) is required to the find the gradient of `f(x, y)` w.r.t. `x` and `y`, which we denote by `∇x` and `∇y` respectively, to be
+
+```@example toy
+(∇x, ∇y) = (A.'y, A * x)
+```
+
+## High-Level Interface
+The high-level interface provides a simple way to "just get the gradients" w.r.t. each argument of `f`:
+```@example toy
+∇x, ∇y = ∇(f)(x, y)
+```
+This interface is implemented in `core.jl`, and is a thin wrapper of the low-level interface constructed above. Here, we first use `∇` to get a function which, when evaluated, returns the gradient of `f` w.r.t. each of it's inputs at the values of the inputs provided.
+
+We may provide an optional argument to also return the value `f(x, y)`:
+```@example toy
+(z, (∇x, ∇y)) = ∇(f, true)(x, y)
+```
+
+If the gradient w.r.t. a single argument is all that is required, or a subset of the arguments for an N-ary function, we recommend closing over the arguments which respect to which you do not wish to take gradients. For example, to take the gradient w.r.t. just `x`, one could do the following:
+```@example toy
+∇(x->f(x, y))(x)
+```
+
+Furthermore, indexable containers such as `Dict`s behave sensibly. For example, the following lambda with a `Dict`:
+```@example toy
+∇(d->f(d[:x], d[:y]))(Dict(:x=>x, :y=>y))
+```
+or a `Vector`:
+```@example toy
+∇(v->f(v[1], v[2]))([x, y])
+```
+
+The methods considered so far have been completely generically typed. If one wishes to use methods whose argument types are restricted then one must surround the definition of the method in the `@unionise` macro. For example, if only a single definition is required:
```

Comment:
Is there a way to run this after method definition (e.g., if the method is in a package you don't control)?

Reply:
Nope, unfortunately not. This is something I've been trying to figure out a decent solution for, but haven't come up with a great one yet. It seems possible that we should be able to write some code that generates a modified version of any particular package that a user wants to `@unionise`, but you really do need to get access to the source somehow.

Reply:
You can get access to the source code of a function with `code_lowered`. I think this might be possible but it might also be A Bad Idea. I'd love to spend some time on that.